e2e-tests/upgrade-sharded/run

```diff
@@ -9,223 +9,223 @@ test_dir=$(realpath $(dirname $0))
+	kubectl_bin apply -f "${src_dir}/deploy/rbac.yaml"
+	kubectl_bin patch deployment "${OPERATOR_NAME}" \
+		-p'{"spec":{"template":{"spec":{"containers":[{"name":"'"${OPERATOR_NAME}"'","image":"'"${TARGET_IMAGE}"'"}]}}}}'
+	kubectl_bin rollout status deployment/"${OPERATOR_NAME}"
+
+	desc 'wait for operator upgrade'
+	until [[ $(kubectl_bin get pods --selector=name="${OPERATOR_NAME}" \
+		-o custom-columns='NAME:.metadata.name,IMAGE:.spec.containers[0].image' \
+		| grep -vc 'NAME' | awk '{print $1}') -eq 1 ]]; do
+		sleep 5
+	done
+	sleep 10
+
+	desc 'check images and generation after operator upgrade'
+	wait_for_running "${cluster}-rs0" "${CLUSTER_SIZE}" "false"
+	wait_for_running "${cluster}-cfg" "${CLUSTER_SIZE}" "false"
+	wait_cluster_consistency "${cluster}" "${CLUSTER_SIZE}"
+	check_applied_images "operator"
+	compare_generation "1" "statefulset" "${cluster}-rs0"
+	compare_generation "1" "statefulset" "${cluster}-cfg"
+	compare_generation "2" "deployment" "${cluster}-mongos"
+	compare_generation "1" "psmdb" "${cluster}"
+
+	desc 'patch psmdb images and upgrade'
+	kubectl_bin patch psmdb "${cluster}" --type=merge --patch '{
         "spec": {
             "crVersion": "'"${TARGET_OPERATOR_VER}"'",
             "image": "'"${TARGET_IMAGE_MONGOD}"'",
             "pmm": { "image": "'"${TARGET_IMAGE_PMM}"'" },
             "backup": { "image": "'"${TARGET_IMAGE_BACKUP}"'" }
         }}'
-    sleep 10
-
-    desc 'check cluster after full upgrade'
-    wait_for_running "${cluster}-rs0" "${CLUSTER_SIZE}" "false"
-    wait_for_running "${cluster}-cfg" "${CLUSTER_SIZE}" "false"
-    wait_cluster_consistency "${cluster}" "${CLUSTER_SIZE}"
-    simple_data_check "${cluster}" "${CLUSTER_SIZE}" 1 "-mongos"
-    check_applied_images "all"
-    compare_generation "2" "statefulset" "${cluster}-rs0"
-    compare_generation "2" "statefulset" "${cluster}-cfg"
-    compare_generation "2" "deployment" "${cluster}-mongos"
-    compare_generation "2" "psmdb" "${cluster}"
-
-    desc 'check if upgrade order is cfg, rs0, mongos'
-    sleep 60 # this is needed because mongos has no status
-    check_upgrade_order "cfg" "${CLUSTER_SIZE}" "1"
-    check_upgrade_order "rs0" "${CLUSTER_SIZE}" "2"
-    check_upgrade_order "mongos" "${CLUSTER_SIZE}" "3"
-
-    desc 'cleanup'
-    kubectl_bin delete -f "${src_dir}/deploy/crd.yaml" || :
-    kubectl_bin delete -f "${src_dir}/deploy/rbac.yaml" || :
-#   kubectl_bin delete pvc --all
-    destroy "${namespace}"
+	sleep 10
+
+	desc 'check cluster after full upgrade'
+	wait_for_running "${cluster}-rs0" "${CLUSTER_SIZE}" "false"
+	wait_for_running "${cluster}-cfg" "${CLUSTER_SIZE}" "false"
+	wait_cluster_consistency "${cluster}" "${CLUSTER_SIZE}"
+	simple_data_check "${cluster}" "${CLUSTER_SIZE}" 1 "-mongos"
+	check_applied_images "all"
+	compare_generation "2" "statefulset" "${cluster}-rs0"
+	compare_generation "2" "statefulset" "${cluster}-cfg"
+	compare_generation "3" "deployment" "${cluster}-mongos"
```

Comment:
why  do we have "3" there ?

Reply:
currently (and previously) operator does not provide `ReadinessProbe.SuccessThreshold` field for mongos deployment (but provides for cfg and rs resources).
here operator allows customise the field and provides default value for it.

in the test, after operator upgrade, it updates `percona.com/last-config-hash` annotation that has the newly added field. btw for next release we will need to revert this changes to pass the test